Assets/Mirror/Editor/Weaver/Readers.cs

```diff
@@ -20,13 +20,27 @@ internal static void Register(TypeReference dataType, MethodReference methodRefe
             readFuncs[dataType.FullName] = methodReference;
         }
 
-        public static MethodReference GetReadFunc(TypeReference variableReference)
+        /// <summary>
+        /// Finds existing reader for type, if non exists trys to create one
+        /// <para>This method is recursive</para>
+        /// </summary>
+        /// <param name="variable"></param>
+        /// <returns>Returns <see cref="MethodReference"/> or null</returns>
+        public static MethodReference GetReadFunc(TypeReference variable)
```

Comment:
```suggestion
        public static MethodReference GetReadFunc(TypeReference typeRef)
```

this is not really a variable,  a variable would be an instance of VariableDefinition, FieldDefinition,  or ParameterDefinition, so I find it very confusing to call this "variable" since all it is is a type.